pkg/db/db.go

```diff
@@ -0,0 +1,118 @@
+package db
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+	"go.uber.org/zap"
+)
+
+func GetConnectionString(host, user, password, name string, port int) string {
+	return fmt.Sprintf(
+		"host=%s port=%d user=%s password=%s dbname=%s sslmode=disable",
+		host,
+		port,
+		user,
+		password,
+		name,
+	)
+}
+
+type Database struct {
+	tables struct {
+		providers string
+	}
```

Comment:
This is a suggestion made by @nb-ohad. @nb-ohad can help you here

Reply:
Why do we need a map (which is not type-safe) if we know exactly what the keys are?

Reply:
- pardon I didn't understand the premise
- Don't we want to just store keys as short form for table names and the type is always a string, isn't it?

Reply:
Let me put it a little bit differently. The DB lib knows exactly which tables it supports and which columns it expects.
This is not a generic lib to handle a database, if we wanted a generic lib I will argue the underlying lib is enough.

As such there is no reason to introduce keys as strings, mainly as this might result in a mistype somewhere and a subtle bug. Accessing `a.b` had less error surface than accessing using `a["b"]`? In the first case, if you mistype b you will get a compile-time error while in the second case you might get a runtime error if you have put some code to check for it.